allow overriding of boilerplate hooks and shell flags in catalog config (#4540)
gruntwork-io/terragrunt: docs-starlight/src/content/docs/02-features/07-scaffold.md

- The `source` URL is configured for you automatically, with the `ref` pointing to the latest "release" tag of the module (found by scanning git tags).
- The `inputs` section is generated for you automatically, and will list all required and optional variables from the module, with their types, descriptions, and defaults, so you can easily fill them in to configure the module as you like.

## Security Features

Terragrunt provides security controls for potentially dangerous template features:

### Shell Functions

Templates can execute shell commands using the `{{shell "command" "args"}}` syntax. This feature is **enabled by default**.

To disable shell functions for security reasons:

```bash
# Via CLI flag
terragrunt scaffold github.com/org/repo//modules/mysql --no-shell

# Via catalog configuration (see below)
```

### Hooks

Templates can define `before` and `after` hooks in their `boilerplate.yml` configuration that execute commands during scaffolding. This feature is **enabled by default**.

To disable hooks for security reasons:

```bash
# Via CLI flag
terragrunt scaffold github.com/org/repo//modules/mysql --no-hooks

# Via catalog configuration (see below)
```

> **Security Warning**: Both shell functions and hooks can execute arbitrary commands on your system. Consider disabling these features when using untrusted templates or for enhanced security.

Reviewer: ```suggestion
**Security Note**: Both shell functions and hooks can execute arbitrary commands on your system. Consider disabling these features for enhanced security. Do not scaffold templates you do not trust.
```

### Configuration Precedence

> When using scaffold with catalog integration, configuration follows this precedence order (highest to lowest):

Reviewer: ```suggestion
When using scaffold via the catalog integration, configuration follows this precedence order (highest to lowest):
```
